transports/upgrades/asb-9.1toasbs-1.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Azure Service Bus Transport Legacy Upgrade Version 9.1 to Azure Service Bus 1.0
+summary: Tips when upgrading Azure Service Bus Legacy transport to Azure Service Bus.
+reviewed: 2020-02-26
+component: ASBS
+related:
+ - transports/azure-service-bus
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 7
+ ---
+
+
+## Migrating from Azure Service Bus (Legacy) Transport to the Azure Service Bus Transport
+
+Endpoints using the Azure Service Bus Transport can be run side-by-side with endpoints using the Legacy Azure Service Bus Transport as long as the [backwards compatibility](/transports/azure-service-bus/compatibility) requirements are met.
+
+## API Differences
+
+There are differences between the API for the Legacy Azure Service Bus Transport and the Azure Service Bus Transports.
+
+### Max lock duration
+
+It is not possible to change the default message lock duration or the auto-renewal for locks.
+
+Azure Service Bus uses a default of 5 minutes for the message lock duration and it does not auto-renew this. It is not possible to change these values as they have been chosen for enhanced reliability and to suppor the most common use-cases. This involves specifying a lock duration of 5 minutes. One of the reasons we chose 5 mins LockDuration is due to the fact there was a lot of confusion about ASB's default (30 seconds), Prefetch, and AutoLockRenewal.
+
+The most common use case for NServiceBus endpoints is to process messages without the need to manage for lock expiration. Therefore the default values for LockDuration and Prefetch in the Azure Service Bus Transport have been selected to ensure no messages lose their locks when processing is longer than 30 seconds. If processing is successful, the message will be ack'd and completed. In case the message fails processing, it will go through recoverability. Only when there's an abnormal shutdown behaviour (e.g. such as an endpoint crashing) would take up-to 5 minutes for the message to re-appear.
```

Comment:
Since the section is about lock duration, let's keep auto-renewal out of here and add it as a separate section?

```suggestion

Azure Service Bus set message lock duration (`MaxLockDuration`) to a maximum of 5 minutes by default. This value has been chosen for enhanced reliability and to support the most common use-cases. One of the reasons we chose to set lock duration to the maximum is to avoid the need to reconfigure the setting when the default time of 30 seconds in the legacy transport was not sufficient for processing and was causing `LockLostException` exceptions. 

The most common use case for NServiceBus endpoints is to process messages without the need to manage for lock expiration. Therefore the default values for LockDuration and Prefetch in the Azure Service Bus transport have been selected to ensure no messages lose their locks when processing is longer than 30 seconds. If processing is successful, the message will be acknowledged and completed. In case the message fails to process, it will go through recoverability. Only when there's an abnormal shutdown behaviour (e.g. such as an endpoint crashing) would take up to 5 minutes for the message to re-appear.

Q: is it possible to change the default message lock duration for an endpoint?
A: yes. The transport is no longer modifying any existing queues (or other entities). Lock duration modification should be done directly on the queue using the [Azure portal](https://portal.azure.com/), [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/servicebus/queue?view=azure-cli-latest#az-servicebus-queue-update), or a 3rd part tool.

Q: would a lock duration set to the maximum time have a negative impact on a message that fails to process in less than 5 minutes?
A: no. The transport will release the message. If it fails to do so, the message will be unlocked automatically by the broker and become available for re-processing.

Q: do I still need to consider the size of the prefetch with the lock duration set to maximum?
A: yes. While the new transport has been build with the most considerable defaults, handlers are still custom code and if take too long, even the optimal defaults can become sub-optimal.
```